packages/provider/src/RelayClient.ts

```diff
@@ -537,6 +546,61 @@ export class RelayClient {
       scoreCalculator
     }
   }
+
+  async _verifyDryRunSuccessful (gsnTransactionDetails: GsnTransactionDetails): Promise<Error | undefined> {
+    // TODO: only 3 fields are needed, extract fields instead of building stub object
+    const dryRunRelayInfo: RelayInfo = {
+      relayInfo: {
+        relayManager: '',
+        relayUrl: '',
+        pctRelayFee: '0',
+        baseRelayFee: '0'
+      },
+      pingResponse: {
+        relayWorkerAddress: constants.DRY_RUN_ADDRESS,
+        relayManagerAddress: constants.ZERO_ADDRESS,
+        relayHubAddress: constants.ZERO_ADDRESS,
+        ownerAddress: constants.ZERO_ADDRESS,
+        minMaxPriorityFeePerGas: '0',
+        maxAcceptanceBudget: '0',
+        ready: true,
+        version: ''
+      }
+    }
+    const relayRequest = await this._prepareRelayRequest(gsnTransactionDetails, dryRunRelayInfo)
+    // note that here 'maxAcceptanceBudget' is set to the entire transaction 'maxViewableGasLimit'
+    const relayCallABI: RelayCallABI = {
+      relayRequest,
+      signature: '0x',
+      approvalData: '0x',
+      maxAcceptanceBudget: this.config.maxViewableGasLimit
+    }
+    return await this._verifyViewCallSuccessful(dryRunRelayInfo, relayCallABI, true)
+  }
+
+  async _verifyViewCallSuccessful (
```

Comment:
the idea was to have dry-run INSTEAD of normal validation, not in addition to it.
(and its OK it is not 100% - there is always another full verification on the relayer)

Reply:
Well, I don't know if it should be instead though. It makes sense to me to check with what you can, prepare real request and then check with that as well. These are just milliseconds if everything is working correctly.

Reply:
I think check after dry-run is completely redundant. 
I agree to leave it now to reduce the complexity of this PR - it can be removed in a future PR
I still think we should rely on the gas estimation in the dry-run: with the above "post-check" of txorigin, it add little to the total gas cost.

Reply:
I could not disagree any more. I see no point in removing the actual real check. There is a ton of checks that are short-circuted in a dry-run mode. I don't like the idea of not doing the actual check very very much.